File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {

        int num1 = numberToCheck;
        int c = 0;
        while(num1>0){
            c++;
            num1/=10;
        }

        num1 = numberToCheck;
        int sum = 0;
        while(num1 > 0){
            int mod = num1 % 10;
            int i = 0;
            int val = 1;
            while(i<c){
                val *= mod;
                i++;
            }
            sum += val;
            num1 /= 10;
        }
        return sum==numberToCheck ? true : false;
    }

}

File: solutions/cpp/armstrong-numbers/1/armstrong_numbers.cpp
#include "armstrong_numbers.h"

namespace armstrong_numbers {
    bool is_armstrong_number(int n){
    
        int t1{n}, t2{n};
        if (n<10){
            return true;
        }
        int c{};
        while(t1>0){
            t1/=10;
            c++;
        }
        int sum{};
        for(int i{}; i<c; i++){
            int r = t2%10;
            int v{1};
            for(int j{}; j<c; j++){
                v *=r;
            }
            sum+=v;
            t2 /= 10;
        }
        if(sum==n)
            return true;
        return false;
    }
}  // namespace armstrong_numbers
